src/tests/ftest/nvme/enospace.py

```diff
@@ -0,0 +1,440 @@
+        #For SCM some % space used for system so it wont be 100% full.
+        if pool_usage['scm'] > 25:
+            self.fail('Pool SCM used percentage should be < 20%, instead {}'.
+                      format(pool_usage['scm']))
+
+    def run_enospace_with_bg_job(self):
+        """
+        Function to run test and validate DER_ENOSPACE and expected storage
+        size. Single IOR job will run in background while space is filling.
+        """
+        #Get the initial DER_ENOSPACE count
+        self.der_enspace_log_count()
+
+        # Start the IOR Background thread which will write small data set and
+        # read in loop, until storage space is full.
+        out_queue = queue.Queue()
+        job = threading.Thread(target=self.ior_bg_thread,
+                               kwargs={"results": out_queue})
+        job.daemon = True
+        job.start()
+
+        #Run IOR in Foreground
+        self.run_enospace_foreground()
+        # Verify the background job queue and make sure no FAIL for any IOR run
+        while not self.out_queue.empty():
+            if self.out_queue.get() == "FAIL":
+                self.fail("One of the Background IOR job failed")
+
+    @skipForTicket("DAOS-4846")
+    def test_enospace_lazy_with_bg(self):
+        """Jira ID: DAOS-4756.
+
+        Test Description: IO gets DER_NOSPACE when SCM and NVMe is full with
+                          default (lazy) Aggregation mode.
+
+        Use Case: This tests will create the pool and fill 75% of SCM size which
+                  will trigger the aggregation because of space pressure,
+                  next fill 75% more which should fill NVMe. Try to fill 60%
+                  more and now SCM size will be full too.
+                  verify that last IO fails with DER_NOSPACE and SCM/NVMe pool
+                  capacity is full.
+
+        :avocado: tags=all,hw,medium,nvme,ib2,full_regression
+        :avocado: tags=der_enospace,enospc_lazy,enospc_lazy_bg
+        """
+        print(self.pool.pool_percentage_used())
+
+        #Run IOR to fill the pool.
+        self.run_enospace_with_bg_job()
+
+    @skipForTicket("DAOS-4846")
+    def test_enospace_lazy_with_fg(self):
+        """Jira ID: DAOS-4756.
+
+        Test Description: Fill up the system (default aggregation mode) and
+                          delete all containers in loop, which should release
+                          the space.
+
+        Use Case: This tests will create the pool and fill 75% of SCM size which
+                  will trigger the aggregation because of space pressure,
+                  next fill 75% more which should fill NVMe. Try to fill 60%
+                  more and now SCM size will be full too.
+                  verify that last IO fails with DER_NOSPACE and SCM/NVMe pool
+                  capacity is full. Delete all the containers.
+                  Do this in loop for 10 times and verify space is released.

```

Comment:
Please be aware that container destroy will release space asynchronously, so it could take some time to release all space after container is destroyed.